### PR TITLE
Added SetTemperature method. Sets the device to white mode.

### PR DESCRIPTION
**Pull request by Aytackydln**

### CODE REVIEW

File: YeeLightAPI/YeeLightAPI/YeeLightAPI.cs [hunk 1]
```diff
@@ -188,6 +188,25 @@ public void CloseConnection()
             return SendCommandMessage(1, "set_rgb", new string[] { value.ToString(System.Globalization.CultureInfo.InvariantCulture), Utils.GetJsonStringFromParamEnum(effectType), duration.ToString(System.Globalization.CultureInfo.InvariantCulture) });
         }
 
+
+        public bool SetTemperature(
```
Comment (SnakePin) [inline YeeLightAPI/YeeLightAPI/YeeLightAPI.cs:192]: Can you add an XML documentation comment to this function?

File: YeeLightAPI/YeeLightAPI/YeeLightAPI.cs [hunk 2]
```diff
@@ -261,7 +280,7 @@ private void ThrowExceptionIfInMusicMode()
                 throw new Exceptions.DeviceIsAlreadyInMusicMode();
             }
         }
-        private bool SendCommandMessage(int id_pair, string method_pair, string[] params_pair)
+        private bool SendCommandMessage(int id_pair, string method_pair, object[] params_pair)
```
Comment (SnakePin) [inline YeeLightAPI/YeeLightAPI/YeeLightAPI.cs:283]: Implicit `ToString()` will have different behaviors on different languages. I suggest changing this back to a string array.

File: YeeLightAPI/YeeLightAPI/YeeLightAPI.cs [hunk 3]
```diff
@@ -188,6 +188,25 @@ public void CloseConnection()
             return SendCommandMessage(1, "set_rgb", new string[] { value.ToString(System.Globalization.CultureInfo.InvariantCulture), Utils.GetJsonStringFromParamEnum(effectType), duration.ToString(System.Globalization.CultureInfo.InvariantCulture) });
         }
 
+
+        public bool SetTemperature(
+            int temperature,
+            int duration = Constants.MinValueForDurationParameter,
+            Constants.EffectParamValues effectType = Constants.EffectParamValues.SUDDEN
+            )
+        {
```
Comment (SnakePin) [inline YeeLightAPI/YeeLightAPI/YeeLightAPI.cs:197]: Can you please add range checks here for the `temperature` and `duration` parameters? The range check for `temperature` should look like this
`ThrowExceptionIfIntArgIsOutOfRange("temperature", temperature, Constants.MinValueForTemperatureParameter, Constants.MaxValueForTemperatureParameter);`

**Comment by SnakePin** [comment]
I can submit a commit with the requested changes if you'd rather that.

**Comment by Aytackydln** [comment]
Alright, done your comments. Let me know if you want other changes

**Comment by SnakePin** [comment]
Thank you for your contribution!

**Comment by Aytackydln** [comment]
Can you publish it on nuget soon? So I can reference it on my aurora project

**Comment by SnakePin** [comment]
Are you going to create a PR to the Aurora project? If not, you can add this repo as a submodule to your project and add it as a reference instead. I'm planning on making a NuGet release after adding some more features so I don't really know how long that'll take.

**Comment by Aytackydln** [comment]
I will probabl not make a PR because project seems dead. I will maintain my own fork, mostly for my personal usage. Aurora has so much performance issues also, it will probably take too long to fix at least what I'm using. I will add this as reference then

**Pull request merged**